dashboard/modules/reporter/reporter_agent.py

```diff
@@ -300,10 +316,40 @@ def _get_all_stats(self):
         }
 
     @staticmethod
-    def _record_stats(stats):
+    def _record_stats(stats, cluster_stats):
         records_reported = []
-
         ip = stats["ip"]
+        redis_address = stats["redis_address"]
+
+        # -- Instance count of cluster --
+        if "autoscaler_report" in cluster_stats and str(ip) == str(
+                redis_address):
+            active_nodes = cluster_stats["autoscaler_report"]["active_nodes"]
+            for node_type, active_node_count in active_nodes.items():
+                records_reported.extend([
+                    Record(
+                        gauge=METRICS_GAUGES["cluster_active_nodes"],
+                        value=active_node_count,
+                        tags={"node_type": node_type})
+                ])
+
+            failed_nodes = len(
+                cluster_stats["autoscaler_report"]["failed_nodes"])
+            logger.info("~~~~~ failed nodes" + str(cluster_stats["autoscaler_report"]["failed_nodes"]))
+            cluster_failed_nodes_record = Record(
+                gauge=METRICS_GAUGES["cluster_failed_nodes"],
+                value=failed_nodes,
+                tags={})
```

Comment:
remove the empty tag={}

Reply:
need the tag now that we are including node type